dsl/properties/scripts/ImportMicroservices.groovy

```diff
@@ -0,0 +1,440 @@
+/**	Docker Compose File structure:
+ *
+ *	version:
+ *	'2'
+ *	services:
+ *   web:
+ *	    image: nginx:13.1
+ *	    ports:
+ *	     - "5000:5000"
+ *	    volumes:
+ *	- .:/code
+ *	redis:
+ *	image: redis
+ */
+
+public class ImportMicroservices extends EFClient {
+	def dockerClient
+    def pluginConfig
+    def clusterEndpoint
+	def composeConfig
+    def importedSummary = [:]
+	
+	static final String CREATED_DESCRIPTION = "Created by Container ImportMicroservices"
+	
+	def ImportMicroservices(def dockerClient, def pluginConfig, def composeConfig) {
+        dockerClient = dockerClient
+        pluginConfig = params.pluginConfig
+        accessToken = kubeClient.retrieveAccessToken(pluginConfig)
+		this.composeConfig = composeConfig
+        clusterEndpoint = pluginConfig.clusterEndpoint
+    }
+	
+	def buildServicesDefinitions(def projectName, def applicationName) {
+		def efServices = []
+		composeConfig.services.each { name, serviceConfig ->
+			def efService = buildServiceDefinition(name, projectName, applicationName, serviceConfig)
+			efServices.push(efService)
+        }
+		efServices
+	}
+	
+	def buildServiceDefinition(def name, def projectName, def applicationName, def serviceConfig) {
+        def efServiceName = name
+        def efService = [
+            service: [
+                serviceName: efServiceName
+            ],
+            serviceMapping: [:]
+        ]
+
+        // Service Fields
+		efService.service.defaultCapacity = serviceConfig.deploy?.replicas
+		efService.service.minCapacity = serviceConfig.updateConfig?.parallelism
+		
+		String[] imageInfo = serviceConfig.image?.split(':')
+        def image = ''
+        def version = ''
+        if (imageInfo && imageInfo.length > 0) {
+            image = imageInfo[0]
+            if (imageInfo.length > 1) {
+                version = imageInfo[1]
+            }
+        }
+		
+		// ENV variables
+        def envVars = ""
+        serviceConfig.environment.entries.each{key, value ->
+            envVars += """
+            environmentVariable '$key', {
+                type = 'string'
```

Comment:
Don't generate DSL at this stage. Just put data into HashMap, e.g. 
```
        container.env = kubeContainer.env?.collect {
            [environmentVariableName: it.name, value: it.value]
        }
```

Reply:
Then this data can be easily saved into EF. And this DSL is not flexible and horrible to read.

Reply:
Done